mesonbuild/interpreter.py

```diff
@@ -2152,6 +2151,8 @@ def run_command_impl(self, node, args, kwargs, in_builddir=False):
             'or configure_file(), or a compiler object; not {!r}'
         if isinstance(cmd, ExternalProgramHolder):
             cmd = cmd.held_object
+            if isinstance(cmd, build.Executable):
+                raise InterpreterException('Cannot use an executable during configuration')
```

Comment:
We need to help our users more here. The error message at least needs to say that the executable was overriden and that as it is built it is not available for execution in the build file.